python/src/scipp/compat/mantid.py

```diff
@@ -108,13 +111,20 @@ def ConvertEventWorkspaceToDataset(ws, load_pulse_times):
             pt = ws.getSpectrum(i).getPulseTimes()
             labs[sc.Dim.Position, i].values = np.asarray([p.total_nanoseconds()
                                                           for p in pt])
-
-    coords_and_labs = {"coords": {dim: coords, sc.Dim.Position: pos},
-                       "labels": {"spectrum_number": num,
-                                  "component_info": comp_info}}
+        w = ws.getSpectrum(i).getWeights()
+        weights[sc.Dim.Position, i].values = w
+        if not contains_weighted_events and (len(w) > 0):
+            if np.amax(w) > 1.0:
+                contains_weighted_events = True
+
+    coords_labs_data = {"coords": {dim: coords, sc.Dim.Position: pos},
+                        "labels": {"spectrum_number": num,
+                                   "component_info": comp_info}}
     if load_pulse_times:
-        coords_and_labs["labels"]["pulse_times"] = labs
-    return sc.DataArray(**coords_and_labs)
+        coords_labs_data["labels"]["pulse_times"] = labs
+    if contains_weighted_events:
+        coords_labs_data["data"] = weights
```

Comment:
If there are weight, then Mantid always also stores errors for the weights, these should be stored as the variances.